### ISSUE TITLE
TFT35 MKS-Robin Nano white Display

### ISSUE THREAD

**Issue by ph1lloops** [makerbase-mks/MKS-Robin#124]
I flashed Marlin 2.0 and on the Robin Nano 1.1 and also the MKS Robin TFT35 V1.0 Firmware for the Display. After Booting the display keeps white and nothing happens. Connecting to the MKS Robin Nano with pronterface works. Do I have to set something else in Marlin?

**Comment by zillarob** [comment]
You dont need fw for the screen, its build into Marlin.
You do need to enable it and the touch buttons in Marlin.
Look for the FMSC stuff down toward the bottom of config.h.

**Comment by ph1lloops** [comment]
Hey,
Thanks for you help. I was searching in Marlin in configuration.h but I can not find anything with FMSC. Maybe you can send me a screenshot of the part you mean.

Thanks!
<Philipp

Am 29.08.2019 um 01:32 schrieb zillarob <notifications@github.com<mailto:notifications@github.com>>:


You dont need fw for the screen, its build into Marlin.
You do need to enable it and the touch buttons in Marlin.
Look for the FMSC stuff down toward the bottom of config.h.

**Comment by DaVincIan** [comment]
Start by in the config.h file search for TFT.

**Comment by ph1lloops** [comment]
<img width="715" alt="Bildschirmfoto 2019-08-29 um 06 04 28" src="https://user-images.githubusercontent.com/36089088/63909434-e7e8c000-ca22-11e9-9f65-8038bce2f54c.png">

**Comment by zillarob** [comment]
I forgot they have an odd version here.
It used to say that, but the newer version is slightly different now.
Id grab a fresh version from the main branch. Only use the ones here as a template for some of the stuff like serial settings etc.

**Comment by ph1lloops** [comment]
Using the fresh version with the nano board selected and the TFT aktivated = display stays completely off

**Comment by zillarob** [comment]
There is no tft option in the main marlin branch.

![nan](https://user-images.githubusercontent.com/42916376/63910010-0636c780-c9f3-11e9-8930-6ef6eee01ecb.JPG)

**Comment by ph1lloops** [comment]
That’s the line changed like in your screenshot. What is also missing in my firmware?

LG Philipp

Am 29.08.2019 um 06:22 schrieb zillarob <notifications@github.com<mailto:notifications@github.com>>:


There is no tft option in the main marlin branch.

[nan]<https://user-images.githubusercontent.com/42916376/63910010-0636c780-c9f3-11e9-8930-6ef6eee01ecb.JPG>

**Comment by zillarob** [comment]
Hard to say.
Ive been playing with it on my Tevo Nereus and the toughest thing I remember was getting the buttons to work.
It does seem a lil buggy though. I cant say if its the fw, the board, or me doing something goofy in the fw though.
It does seem like if I dl and setup a new it will be fine, but as time goes by and I turn stuff on or off it just gets worse and worse.

**Comment by zillarob** [comment]
You could try adding this if not there
![nanpin](https://user-images.githubusercontent.com/42916376/63910408-945f7d80-c9f4-11e9-8fef-5b0abb696a4d.JPG)

**Comment by ph1lloops** [comment]
Wow thanks for your help. I’ll try that when I am back home. Just add that to the end of configuration.h right? Just to make sure: I changed the Board and the Ports + uncommented the TFT line like in your screenshot. Anything else I have to take care of?

LG Philipp

Am 29.08.2019 um 06:33 schrieb zillarob <notifications@github.com<mailto:notifications@github.com>>:


You could try adding this if not there
[nanpin]<https://user-images.githubusercontent.com/42916376/63910408-945f7d80-c9f4-11e9-8fef-5b0abb696a4d.JPG>

**Comment by zillarob** [comment]
You need to see what is already in there, and then add the stuff that isnt. 
Dont just paste that at the bottom.

I dont know what you will have to change for your printer, but that is what I use and the screen and touch buttons work.

**Comment by ph1lloops** [comment]
Yes sure I’ll change all the stuff for my printer but first I wanted to make the display to work.

LG Philipp

Am 29.08.2019 um 06:50 schrieb zillarob <notifications@github.com<mailto:notifications@github.com>>:


You need to see what is already in there, and then add the stuff that isnt.
Dont just paste that at the bottom.

I dont know what you will have to change for your printer, but that is what I use and the screen and touch buttons work.

**Comment by alfrankgit** [comment]
You need to uncomment this in PINS_MKS_ROBN_NANO.h so that that 3.5“ display gets properly initialized:

//#define MKS_ROBIN_TFT35     //Uncomment when using the robin tft35 display

**Comment by ph1lloops** [comment]
Is that in the configuration.h that line?

LG Philipp

Am 29.08.2019 um 18:18 schrieb alfrankgit <notifications@github.com<mailto:notifications@github.com>>:


You need to uncomment this in PINS_MKS_ROBN_NANO.h so that that 3.5“ display gets properly initialized:

//#define MKS_ROBIN_TFT35 //Uncomment when using the robin tft35 display

**Comment by alfrankgit** [comment]
It is in the file PINS_MKS_ROBN_NANO.h as I wrote above.

**Comment by ph1lloops** [comment]
Yay it works now. The only thing: the TFT shows the Marlin layout. What do I need to change to get the MKS TFT design?
Big big thanks!
![IMG_0039](https://user-images.githubusercontent.com/36089088/63970091-97fe0d80-caa3-11e9-8271-eab257ed0c34.jpeg)

**Comment by zillarob** [comment]
I dont think you can.
You either get an updated version of marlin with the marlin menus,
or an old version of marlin with mks menus and who knows wtf else going on because mks never released the source. (if you look at it with a hex editor it was way old, like 1.0)

That version you are using from here is getting pretty old now also (2.0 at least), and the touch buttons arent nearly as responsive as the current main branch iirc.

**Comment by ph1lloops** [comment]
Ah ok. I was looking for the 
define MKS_ROBIN_TFT35 //Uncomment when using the robin tft35 display
in the newest 2.0.x Marlin firmware, but I can not see it there. How can I activate the TFT35 in that newer Marlin version?
Thanks again for your help and work!

**Comment by alfrankgit** [comment]
No, the MKS style menu is only in the precompiled Release 1.0.2, there’s no source code for that.
I think the Marlin 2.0 from MKS is just for alibi.

**Comment by ph1lloops** [comment]
Ok I flashed the newest marlin and the display runs after uncommenting 
//#define LCD_RESET_PIN      PF6

The touch button are missing and the resolutions seems to be not correct. These two things and i am happy. Do have an idea how to fix those last two things?

**Comment by zillarob** [comment]
Never seen that res thing.
Enable the touch buttons and maybe the other stuff I mentioned above.

**Comment by zillarob** [comment]
Chip on nano doesnt have PF6. I think that was supposed to be PC6 for nano.

**Comment by Technology89** [comment]
Tried the same as Phillip today and got exactly the same result. Display just shown in odd resolution and no touch support.

**Comment by 3Dangel** [comment]
I would compare the pins assignment and how it is configured in mks-Robin
and verify the pinout from the schematic.
Mks-Robin touch is working.

Check the issue case in mks-robin ( use key word on Google “ mks Robin
Marlin”)

A gentlemen posted a small FW to test what LCD controller the board is
using.

It seemed to me that for the moment, there is only couple lcd controller
that are supported.

Hope this helps.

**Comment by zillarob** [comment]
I dont know if this is what it is using, but I see quite a few changes in the tft upscaler thing in lcd dogm between the older one I am using and the current version.
If I get bored this eve, I will do up a freshy and see if I see the same.

**Comment by Technology89** [comment]
Got the touchbuttons to show up but they don't work. Didn't define TOUCH_BUTTONS in configuration.h



https://i.imgur.com/yTqXn7Lr.jpg

**Comment by 3Dangel** [comment]
Yours is the latest version that they just forked to.
Mine worked with the green original settings.

Thanks for advise as I am about to upload the newest updated FW as yours.
Color is good!!!!
Cheers

**Comment by 3Dangel** [comment]
Hello Technology89,

 Would it be possible to get zip file from you?
I’m having issue with my just downloaded version of Marlin.
The old one that I had for s month or so still compiles correctly ( without
your color buttons)

Thanks in advance.
Bob

**Comment by Technology89** [comment]
Gonna upload tomorrow. But remember that the buttons not working. They are just shown but not touchable.

**Comment by 3Dangel** [comment]
Thanks man…
Will compare what I have
Compiling now with my setttings..

Thanks v much!!!
Robert

Sent from Mail for Windows 10

**Comment by Technology89** [comment]
Here you go. Maybe you need to tune some settings as i am using LV8729 for Z and E instead of the a4988

https://github.com/Technology89/Marlin_2.0_Sapphire_Pro

**Comment by 3Dangel** [comment]
Thanks!!
I got it last night to work my touch.
Not from the latest Marlin.
No color, just green standard.
And SD works as well.
It was a heck of work an try on error!!

**Comment by Technology89** [comment]
But still marlin 2.0? SD eeprom also working? Mind sharing your version?

**Comment by 3Dangel** [comment]
mine is the Robin
not sure about eeprom
only got it to load and checked my menus.

I will post like your did in github tonight

but short thing for your to try is the Touch_buttom calibration

those values are most likely off

these are my values

 #define XPT2046_X_CALIBRATION   -11876
  #define XPT2046_Y_CALIBRATION  8755
  #define XPT2046_X_OFFSET       346
  #define XPT2046_Y_OFFSET        -20
kind of flipped from what it is on Marlin.

I did use a calibration tool from a gentleman who worked extensively for
Robin...

and my version is older ..
the new one gives me black screen, so it is something not right.....
need to check yours to compare since the NANO is very similar to my
Robin.....

cheers
keep in touch

(working with another gentleman from Sought Africa as well)
me from Calgary, AB Canada

Robert

**Comment by Technology89** [comment]
Gonna try these values tomorrow but you have a '#' in front of these values so they are not assigned

**Comment by 3Dangel** [comment]
It is all in one line
 #define XPT2046_X_CALIBRATION   -11876

**Comment by 3Dangel** [comment]
I see what you are talking about

You need the “#”

It is the “//“ that uncomment the line

**Comment by Technology89** [comment]
Of course... Seems like i had a brainfart 😄

**Comment by zillarob** [comment]
Finally got to try a fresh version out, looks ok.
Really like the new cancel button. Saves a bunch of time in the menus.

I think the toughest part on the new versions is the pins file. This is one I have been using for a while and seems to be ok, but have been being lazy and just rolling stuff into it lately. Not sure how much is actually needed in the new stuff or if there is some redundancy, but I should go back and start fresh one of these days and find out how much is needed.
This is for Main branch Marlin after you setup config and config.h for whatever you have.
Might need to change a few pins also, I run BLTouch so some stuff there is swapped around.
[pins_MKS_ROBIN_NANO.zip](https://github.com/makerbase-mks/MKS-Robin/files/3572664/pins_MKS_ROBIN_NANO.zip)

**Comment by zillarob** [comment]
The X on the left is nice. 
No more having to scroll up to the top to go to the previous menu, just hit the X.

![20190904_000639](https://user-images.githubusercontent.com/42916376/64225250-8ef8ac00-cea8-11e9-8ff1-480d51993546.jpg)

**Comment by 3Dangel** [comment]
Cool staff!!!
I was not able to make my Robin v2.1 to work yet under Marlin2 xxxbug yet.
I got it to compile and get the .bin but the screen stays black.

Thinking that it has to be pins issues.
Thanks for sharing yours.
Mine “works” ( not hooked up yet with motors and etc) under an older
brancknofnMarlin

**Comment by zillarob** [comment]
Im on a Nano so no idea how it will work on the other boards.

**Comment by Technology89** [comment]
@zillarob can you upload the whole marlin? Your screen is fullscreen and not only the left corner like with mine version. I don't think this is only a pins issue on mine.

**Comment by zillarob** [comment]
Doesnt look like, too big.

**Comment by zillarob** [comment]
Here is the compiled one if you want to test, but dont try to print with it.
[Robin_nano.zip](https://github.com/makerbase-mks/MKS-Robin/files/3572767/Robin_nano.zip)

**Comment by 3Dangel** [comment]
Zillarob,

Not sure if it is too much too ask:
Would it be possible to have the marlin config (x2) and thenplatformIO.ini
from you?

I know mine is not nano.
But they both share much in common. So I could compare the two and
hopefully I could find the way out  of it.

Thanks in advance

**Comment by zillarob** [comment]
Only thing in pio.ini is setting the env to robin nano.
[configs.zip](https://github.com/makerbase-mks/MKS-Robin/files/3572795/configs.zip)

**Comment by 3Dangel** [comment]
thanks a lot!
it all due to the lib dependency that I saw I am having...
the TMC was givcing me tons of issues and I had to delete some of the
source and replace it with the Bigtree's TMC lib to get the compile going.

**Comment by Technology89** [comment]
You can upload here on github 👌

**Comment by Technology89** [comment]
Thank you so much @zillarob ! The touchbuttons are working but my display is still only small in the top left corner while the touch buttons are sensitive where they used to be...

Can you upload your u8g_dev_tft_320x240_upscale_from_128x64.cpp ? Seems like this one is also different...

**Comment by zillarob** [comment]
That file is untouched from the main marlin branch.

**Comment by Technology89** [comment]
Strange... The touchscreen is scaled correct only the lcd is scaled wrong. It just shown as 128*64 pixels. Also the sd card eeprom won't save. Gonna have a second look at your configuration.h later

**Comment by zillarob** [comment]
I have been messing with the sd stuff. Not positive where I left off, but there was an eeprom.dat on there when I flashed it.

**Comment by zillarob** [comment]
I think I remember seeing that they used a few different chips on the tft. Maybe yours is different?
I never had to mess with it so not if needed to tell it which one or something.

**Comment by zillarob** [comment]
Ya, have eeprom.dat back on there so gonna be you config or config.adv.
I delete it before I flash.

**Comment by 3Dangel** [comment]
cool man!
if there was an issue and you fixed it, worth to post it to the pull on
marlin so it gets merged into it!!!!
What was the setting you made to have it to work?
cheers

**Comment by zillarob** [comment]
The eeprom thing is just a standard marlin setup thing.
When you compare my config and config.adv you will prob find it.

**Comment by 3Dangel** [comment]
ok got it!!
been working to get the screen .....
not sure what they did  but mine is either on booting or white....
well time to sleep.... thanks again!!!!

**Comment by Technology89** [comment]
@3Dangel when your screen is white uncomment '#define LCD_RESET_PIN      PF6'. Then it should work.

@zillarob gonna try your .bin file today and will test if my lcd scales up.

**Comment by zillarob** [comment]
It wont compile (at least mine never did) with PF6 uncommented, the f103vet6 chip doesnt have that pin.

**Comment by Technology89** [comment]
I mean commend out. On my it will compile not matter if LCD_RESET_PIN is active or not. I also had the screen white and the removed the Reset Pin and it worked

**Comment by Technology89** [comment]
I can't test your bin file @zillarob because you have the LCD_RESET_PIN defined which does not work on my panel. Screen is only white.

But i got the Eeprom working with your pins file. I only copied the lcd part before but with everything it works now. So only issue left is the upscaling

**Comment by DaVincIan** [comment]
Hi Everyone

I'm posting here as from what I can see it is the most active area for the subject matter. I have been following the posts and hope to start implementing the lessons learned (so to speak). 

I want to start with some requests:
1) Please add sufficient detail to your post. It is many time really cryptic, especially for users that are not that familiar with the programming environment. 
2) Can we clarify which "source" of Marlin FW "base" code we are referring to. There are a lot of them. There are the Marlin FW here at Makerbase-mks/MKS-Robin, with a branch for each model of the Robin series of controllers. And then there is from the "official" MarlinFirmware/Marlin/tree. Here we tend to use Marlin 2.0x, and configure it for the relevant Robin model.

Technology89, 3DAngel, ZillaRob et al post/uploaded code (according to there posts), but where precise?

So, my problem is somewhat unique, as it seems I'm more or less the only one working with a Robin2 controller board. So far I experience all the symptoms mentioned above. I believe I'm making headway with my pins declaration. I'm having difficulty with setting the classSPI to SPI(2) as this declaration does not exist for the Robin2 environment  (C:\User\.platformio\...)

I can compile without errors, but the flash to board runs to 100% then hangs there, with just a green bar across the TFT screen.

All for now.

**Comment by zillarob** [comment]
@tech Did you try that compiled one? Dont try to do anything other than power it on and push the buttons to look at the menus. Im sure movement will be all screwed up so dont try to move.

That PF6 thing is jacked up for the nano, not sure where they got that from. The chip on the nano doesnt even have that pin and wont compile if you have it set. PC6 should work though.
You do have a Nano right? Not one of the other versions of Robin?

**Comment by Technology89** [comment]
> @tech Did you try that compiled one? Dont try to do anything other than power it on and push the buttons to look at the menus. Im sure movement will be all screwed up so dont try to move.
> 
Yes i tried the compiled one. Like i said my display don't post when LCD_RESET_PIN is set. I only see a white screen. I would need a compiled without the LCD_RESET_PIN

> That PF6 thing is jacked up for the nano, not sure where they got that from. The chip on the nano doesnt even have that pin and wont compile if you have it set. PC6 should work though.
> You do have a Nano right? Not one of the other versions of Robin?
Like i said. Mine compiles without issues but my screen turns white wenn LCD_RESET_PIN is declared. And yes i have a nano.

**Comment by zillarob** [comment]
@DaVi 

I use (and referring to here) a Nano and main marlin branch. The version here is quite old and clunky.

I thought the bigger Robin boards were better sorted already? What we are using has been adapted from that. Seems like the hard way to try to adapt back?

**Comment by zillarob** [comment]
@tech 
Interesting.
I cant remember where I saw the thing about different controller chips on the tft, but you must have one of them. A few people on the Nereus group have played with this and none have had lcd probs.

**Comment by Technology89** [comment]
> @tech
> Interesting.
> I cant remember where I saw the thing about different controller chips on the tft, but you must have one of them. A few people on the Nereus group have played with this and none have had lcd probs.

Thats why they mention it in the pins file:
'/**

Note: MKS Robin TFT screens use various TFT controllers.
If the screen stays white, disable 'LCD_RESET_PIN'
to let the bootloader init the screen.
*/'

Would you like to compile for me without LCD_RESET_PIN active so i can test if scaling works.

**Comment by 3Dangel** [comment]
The PF6 is for Robin
Te PC6 is for nano
You would need to change that on tour board’s Pin setting.

**Comment by Technology89** [comment]
I updated my repo. On this one everything is working except the lcd scaling. EEPROM and touch is working. The configuration.h is for the sapphire pro.
https://github.com/Technology89/Marlin_2.0_Sapphire_Pro

**Comment by zillarob** [comment]
How hard is it to get to the back of the tft and see what chip is on there?
Mine is loose so easy to compare notes.

**Comment by Technology89** [comment]
Need to open the buttom plate. Can't do right now as it is printing for the next 12 hours ^^

**Comment by Technology89** [comment]
@zillarob here is a photo of my screens back:
https://photos.app.goo.gl/822wsYy4qBmmQ4RS8

**Comment by zillarob** [comment]
I believe? that HR2046 chip is the one we are talking about, and same chip on mine.
Strange yours is different. Maybe that is the digitizer chip and another graphics chip hiding somewhere?

**Comment by Technology89** [comment]
If so it must be hidden under the screen...

**Comment by 3Dangel** [comment]
Why I have the feeling that the LCD TFT touch screen with ribbon cables do
not have any processor in them!?
I always thought that they are driven by the main board’s STM32

**Comment by Technology89** [comment]
They are using a serial connection so there must be some type of logic 🤷🏻‍♂️

**Comment by 3Dangel** [comment]
Well that little chip IS the controller

**Comment by zillarob** [comment]
Thats crazy yours does the res thing.
Maybe measure yours diag corner to corner and will do the same. 
Maybe I dont have the size I think I do? =D

**Comment by Technology89** [comment]
Its the tft35 so 3.5 inch

**Comment by zillarob** [comment]
Did you actually measure to make sure? Mine measures right at 3.5"
Really is odd yours behaves so differently.

**Comment by Technology89** [comment]
Edge to edge my caliper shows 3.536 inches. It would be helpful to have a complete marlin or a compiled on with LCD_RESET_PIN disabled to test if it's an configuration issue or because of different hardware

**Comment by DaVincIan** [comment]
All screen sizes, TV etc are measured diagonally.

**Comment by Technology89** [comment]
@DaVinclan this is what i mean with edge to edge.

**Comment by zillarob** [comment]
Looking at the data sheet for the hr2046 looks like it is the driver for the digitizer, not the tft.
There has to be another chip on there somewhere for the tft.

**Comment by 3Dangel** [comment]
LCD-TFT display controller (LTDC), the STM32 MCUs allow to directly drive
high-resolution display panels without any CPU intervention.

**Comment by zillarob** [comment]
I aint gonna clain to know anything about them, but there are prob a lot more pixels than wires in that ribbon cable. Gotta be something in between the stm32 and all those pixels.

**Comment by 3Dangel** [comment]
Me too.
I’m no expert. Just a google searcher.
But from what I see through the Merlin codes, it seems to be that way.
The schematic shows ribbon pins are actually interacting with the STM32
directly and that the STM32 chip has internal TFT-LCD driver.

**Comment by novio8** [comment]
I know that this is off topic, but ...I have White display on new Sapphire Pro I can't or I don't know how to install original firmware. Any Ideas ? thank you

**Comment by zillarob** [comment]
@novio8 
Not off topic at all, exactly the stuff tech is struggling with.

Comment out the reset pin and see if comes on and the resolution is right. 
You prob have the same chip as he does and going to run into the same probs though.

**Comment by alfrankgit** [comment]
@novio8 I posted a complete Sapphire Pro firmware at the site below as I did exactly the same as you and had a hard time to get Two Trees to give me all the files. But when they did it was a new version, 1.0.3, but the bitmaps from 1.0.2. I then renamed and edited some of them so that it all matches up. And it also contained a german luanguage option which I optimized since I am German.
I won’t post the files here because I think that MKS owes us the 1.0.3 binary. But you can get them here on our german 3D printer forum:
https://drucktipps3d.de/forum/topic/schon-gesehen-sapphire-s-pro/?part=18#postid-47373
No warranties! Use at your own risk! No way back!
But ph1lloops already had success with it.

**Comment by novio8** [comment]
Thank you for the link
I have installed your Firmware, everything works just fine, except I'm getting ERR7 while printing.
Did I made a mistake or there is something wrong with Firmware or Sapphire PRO ?

**Comment by alfrankgit** [comment]
I had that too and it seems to be a temperature runaway or similar. Probably caused by the too powerful heater cartridge (50 W). To fix that I increased some values of the temp hysteresis in th cfg file.
#======================== Thermal Runaway Protection =======================
>THERMAL_PROTECTION_PERIOD 40 #Seconds
>THERMAL_PROTECTION_HYSTERESIS 4 #Degrees Celsius
#AFR was 20
>WATCH_TEMP_PERIOD 80 #Seconds
#AFR was 2
>WATCH_TEMP_INCREASE 8 #Degrees Celsius
>THERMAL_PROTECTION_BED_PERIOD 20 #Seconds
>THERMAL_PROTECTION_BED_HYSTERESIS 2 #Degrees Celsius
>WATCH_BED_TEMP_PERIOD 60 #Seconds
#AFR was 2
>WATCH_BED_TEMP_INCREASE 4 #Degrees Celsius

**Comment by novio8** [comment]
alfrankgit,
I'll put Silicone Sock on the heater Block, maybe this will help.
I had two layer shifts printing Benchie, both at approximately 32 mm.
Did yo have layer shifts too with Sapphire PRO ?

**Comment by Technology89** [comment]
> alfrankgit,
> I'll put Silicone Sock on the heater Block, maybe this will help.
> I had two layer shifts printing Benchie, both at approximately 32 mm.
> Did yo have layer shifts too with Sapphire PRO ?

Check the voltage on your drivers. For X and Y they should be around 1.1-1.2V. Mine were at 1.5V which resulted in heavy layershifts.

**Comment by novio8** [comment]
> Check the voltage on your drivers. For X and Y they should be around 1.1-1.2V. Mine were at 1.5V which resulted in heavy layershifts.
OK I will, thankx

**Comment by Technology89** [comment]
@zillarob could you please provide a .bin file with 'LCD_RESET_PIN' deactivated? This would be very helpful to see if my display would scale up correctly.

**Comment by novio8** [comment]
do you mean robin_nano.bin file ?

**Comment by zillarob** [comment]
My guess is it will do the same, but here is the one I use, with the reset pin commented out.

I just picked up a used printer that should come with a different tft.
The metal screen retainer gizmo that originally came on mine was black.
I made the mistake of lending it out for somebody to do some diag and dbag decided to keep it instead of send it back.
The one I bought to replace it has a shiny silver metal retainer gizmo. Maybe a newer/older version?
Will compare when it gets here and see if any differences.

[Robin_nano.zip](https://github.com/makerbase-mks/MKS-Robin/files/3610780/Robin_nano.zip)

**Comment by Technology89** [comment]
Thanks, gonna try the next days 👍

**Comment by Technology89** [comment]
Okay it's the same. Image only in top left corner.

**Comment by zillarob** [comment]
That other tft showed up and works the same as my other one does.
There was a version number on there that I cant remember of the top of my head, but they were the same. Only difference was one had a black metal retainer for the screen, the other had a shiny chrome-like one.

**Comment by pcopa** [comment]
I also had the dreaded white screen, but after a lot Research and help from the German forum 
https://drucktipps3d.de/forum/topic/schon-gesehen-sapphire-s-pro/?part=24 , I was able to compile a set of files that reset the Sapphire-Pro printer to the latest revision (for me).

1. Just add the unzipped files to an empty micro-SD card.
2. Switch printer off
3. Insert micro-SD card
4. Switch printer back on and let the update do its work. 

[TFT35_V1.0.2 Sapph-Pro191002.zip](https://github.com/makerbase-mks/MKS-Robin/files/3683955/TFT35_V1.0.2.Sapph-Pro191002.zip)

**Comment by sobieh** [comment]
Hello.
Any idea where can i download the original mks firmware (closedsource) v1.0.3 ?
My printer was rerporting such version in About and i would like to be able to restore it after failed marlin 2,0 attempts.

**Comment by 3Dangel** [comment]
Did you try the 2.03?
It might be compatible.
Alternatively, you may want to try the branch of Marlin by jmz52

But the one at GitHub from MKS would work.

What are the things that it did not work?

**Comment by 3Dangel** [comment]
I found V1.02 in my gmail folder.
If you still need it, let me know

**Comment by zillarob** [comment]
Id throw an SKR v1.3 or even a Gen L in long before I put that awful fw back on.

I dont know if you use e1 for autofan, but I played hell finding one that had it enabled again after playing with fw from this repo.

**Comment by 3Dangel** [comment]
I agree with you.
Old Robins are very limited.

But I still run it on v1.03 for my Zonestar M802 acrylic frame.
And has not failed me yet.

I have another Robin v 2.1 that either will be running on Marlin2 ( cross
finger that most of the functionality will run)
Or just buy a SKR 1.3 ( but still have 2 MKs-Sbase V1.3 in the box for
years now.

**Comment by zillarob** [comment]
SKR v1.3 will be drop in for a Nano (maybe Mini too), same size and screws.
Id throw BigTreeTech some love because they do good stuff and have always released info on their products.
All the info that MKS has released recently is trying to catch up to them. 
If you compare the 2 githubs, its like MKS used the BTT github as a template =D

**Comment by pcopa** [comment]
@sobieh have you ried my

TFT35_V1.0.2 Sapph-Pro191002.zip
It got me back to the original factory settings.
![IMG_20191002_172928](https://user-images.githubusercontent.com/15312957/66248062-8d611480-e6f0-11e9-9649-f07dd5f1735c.jpg)

**Comment by napiglet** [comment]
Hello.. 
How can I add a Korean font? Can you share the code?
The environment I'm testing... mks robin nano v1.2 / mks tft35
![image](https://user-images.githubusercontent.com/74293699/123505783-feeaae80-d69b-11eb-9163-3aee183f4421.png)

**Comment by napiglet** [comment]
> @sobieh have you ried my
> 
> TFT35_V1.0.2 Sapph-Pro191002.zip
> It got me back to the original factory settings.
> ![IMG_20191002_172928](https://user-images.githubusercontent.com/15312957/66248062-8d611480-e6f0-11e9-9649-f07dd5f1735c.jpg)

**Comment by napiglet** [comment]
> TFT35_V1.0.2 Sapph-Pro191002.zip
> It got me back to the original factory settings.
> ![IMG_20191002_172928](https://user-images.githubusercontent.com/15312957/66248062-8d611480-e6f0-11e9-9649-f07dd5f1735c.jpg)

**Comment by lucas83-lee** [comment]
> Hello..
> How can I add a Korean font? Can you share the code?
> The environment I'm testing... mks robin nano v1.2 / mks tft35
> ![image](https://user-images.githubusercontent.com/74293699/123505783-feeaae80-d69b-11eb-9163-3aee183f4421.png)

i think it doesn't support korean language.
